packages/tslint-config/src/rules/blueprintClassesConstantsRule.ts

```diff
@@ -29,15 +28,136 @@ export class Rule extends Lint.Rules.AbstractRule {
+            // tslint:disable-next-line:no-conditional-assignment
+            while ((currentMatch = BLUEPRINT_CLASSNAME_PATTERN.exec(fullText)) != null) {
+                const fullBlueprintName = currentMatch[1]; // e.g. pt-breadcrumb
+                const blueprintClassName = getBlueprintClassName(fullBlueprintName); // e.g. breadcrumb
+
+                // See if we should ignore this class or not.
+                if (blueprintClassName == null || shouldIgnoreBlueprintClass(blueprintClassName)) {
+                    continue;
+                } else {
+                    ptMatches.push(fullBlueprintName);
+                }
+            }
+            if (ptMatches.length > 0) {
+                // we have to fail the entire node so that we can make multiple replacements, if necessary
+                ctx.addFailureAtNode(
+                    node,
+                    Rule.FAILURE_STRING,
+                    getReplacement(node, ptMatches, ctx.sourceFile, shouldFixImports),
+                );
+                shouldFixImports = false;
+            }
+        }
+
+        return ts.forEachChild(node, callback);
+    }
+}
+
+function getReplacement(
+    node: ts.StringLiteralLike | ts.TemplateExpression,
+    ptClassStrings: string[],
+    file: ts.SourceFile,
+    shouldFixImport: boolean,
+) {
+    const replacements: Lint.Replacement[] = [];
+
+    // We may need to add a blueprint import to the top of the file. We only want to do this once per file, otherwise,
+    // we'll keep stacking imports and mess things up.
+    if (shouldFixImport) {
+        replacements.push(addImportToFile(file, ["Classes"], "@blueprintjs/core"));
+    }
+
+    if (utils.isStringLiteral(node)) {
+        // remove all illegal classnames, then slice off the quotes, and trim any remaining white space
+        const stringWithoutPtClasses = ptClassStrings
+            .reduce((value, cssClass) => {
+                return value.replace(cssClass, "");
+            }, node.getText())
+            .slice(1, -1)
+            .trim();
+        const templateStrings = ptClassStrings.map(n => `\${${convertPtClassName(n)}}`).join(" ");
+        if (stringWithoutPtClasses.length > 0) {
+            const replacement = `\`${templateStrings} ${stringWithoutPtClasses}\``;
+            replacements.push(
+                new Lint.Replacement(node.getStart(), node.getWidth(), wrapForParent(replacement, node, node.parent)),
+            );
+        } else {
+            if (ptClassStrings.length === 1) {
+                const replacement = convertPtClassName(ptClassStrings[0]);
+                replacements.push(
+                    new Lint.Replacement(
+                        node.getStart(),
+                        node.getWidth(),
+                        wrapForParent(replacement, node, node.parent),
+                    ),
+                );
+            } else {
+                const replacement = `\`${templateStrings}\``;
```

Comment:
this could be combined with line 97 since the `replacements.push()` is repeated.

```js
const replacement = ptClassStrings.length === 1 ? convertPtClassName(ptClassStrings[0]) : `\`${templateStrings}\``
```